sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputControlTrackingUnit.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl;
+
+import com.azure.cosmos.implementation.OperationType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ThroughputControlTrackingUnit {
+
+    private static final Logger logger = LoggerFactory.getLogger(ThroughputControlTrackingUnit.class);
+
+    private final OperationType operationType;
+    private final AtomicInteger rejectedRequests;
+    private final AtomicInteger passedRequests;
+    private final AtomicReference<Double> successRuUsage;
+    private final AtomicInteger successResponse;
+    private final AtomicInteger failedResponse;
+    private final AtomicInteger outOfCycleResponse;
+    private String throughputControlCycleId;
+
+    public ThroughputControlTrackingUnit(OperationType operationType, String throughputControlCycleId) {
+        this.operationType = operationType;
+
+        this.rejectedRequests = new AtomicInteger(0);
+        this.passedRequests = new AtomicInteger(0);
+        this.successRuUsage = new AtomicReference<>(0d);
+        this.successResponse = new AtomicInteger(0);
+        this.failedResponse = new AtomicInteger(0);
+        this.outOfCycleResponse = new AtomicInteger(0);
+        this.throughputControlCycleId = throughputControlCycleId;
+    }
+
+    public void reset(String newCycleId) {
+        if (this.rejectedRequests.get() > 0
+            || this.passedRequests.get() > 0
+            || this.successResponse.get() > 0
+            || this.failedResponse.get() > 0) {
+
+            double sAvgRuPerRequest = 0.0;
+            if (this.successResponse.get() != 0) {
+                sAvgRuPerRequest = successRuUsage.get() / this.successResponse.get();
+            }
+
+            logger.debug(
```

Comment:
slf4j logger placeholder is preferred over String.format as it helps to avoid paying additional cost when not needed.

please replace String.format with slf4j logger placeholder everywhere.

Reply:
got it, updated